authz/tests/fga_test.go

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright 2023 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package tests runs the fga CLI tests against each file in this directory.
+// It assumes the `fga` binary is in the PATH, e.g. from `make bootstrap`
+package tests
+
+import (
+	"errors"
+	"os/exec"
+	"path/filepath"
+	"testing"
+)
+
+func TestFGA(t *testing.T) {
+	t.Parallel()
+	files, err := filepath.Glob("*.tests.yaml")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(files) == 0 {
+		t.Fatal("no test files found")
+	}
+	for _, f := range files {
+		file := f
+		t.Run(file, func(t *testing.T) {
+			t.Parallel()
+
+			output, err := exec.Command("fga", "model", "test", "--tests", file).CombinedOutput()
```

Comment:
We don't do `make boostrap` before the unit tests. We'd need to start doing that, or run the FGA tests as a separate CI job or test unit.

Reply:
It is cleaner, but make bootstrap is also very time consuming. What about separating the FGA tests and have a dedicated target that does the docker invocation?

Reply:
There's no docker invocation here, just the CLI process.

Oh, but I could actually just import the command code... let me try that.

Reply:
Okay, I've fixed this by actually importing the FGA CLI bits and invoking them directly, so we don't even need `bootstrap` anymore.

The `fga_test.go` still doesn't realize that it depends on `minder.fga`, so it will use cached results if you make changes, but that seems like a minor thing that we can figure out later.